src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -72,6 +72,25 @@ public override async Task<ScriptSecrets> ReadAsync(ScriptSecretsType type, stri
             return type == ScriptSecretsType.Host ? await ReadHostSecrets() : await ReadFunctionSecrets(functionName);
         }
 
+        private static bool TryCreateTokenCredential(string clientId, string clientSecret, string tenantId, out TokenCredential credential)
+        {
+            if (!string.IsNullOrEmpty(clientSecret) && !string.IsNullOrEmpty(tenantId))
+            {
+                credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
+                return true;
+            }
+            else if (!string.IsNullOrEmpty(clientId))
```

Comment:
Here, I think we should just rely on folks/pipelines setting the `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, and `AZURE_FEDERATED_TOKEN_FILE` environment variables. `WorkloadIdentityCredential` will be use these in the absence of the options params.

Reply:
I'm not a big fan of relying on only environment variables. What if the customer has multiple identities in their application? How do they differentiate the environment variables between them?

Reply:
I very much agree with that in general. The primary motivation here is to enable integration test pipeline changes where we do not necessarily need to expand surface area through our own settings. Those standard env vars should be present. I was hoping to keep this more constrained if possible.

However, there is potentially greater risk if we rely on a contract from a dependency which could at some point change beneath us. That has happened before with this component. I could see an argument that, since the client ID and tenant ID are already captured, we'd really just need to get the federated token file bit handled. If that's a change we'd like to see, I can put together a new iteration. CI will need to be updated to remap to our new value.

(This actually also made me realize there's a missing client ID check in the first if block, so additional iteration required regardless)

Reply:
Ah if this is for integration tests only, I am fine with a limited support for now and then we can revisit and expand upon this behavior in the future.

Is there a feature flag or environment variable we can hide workload identity support behind, so it is not unintentionally used outside of integration tests?